packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -147,12 +144,35 @@ export interface PluginContainer {
 type PluginContext = Omit<
   RollupPluginContext,
   // not documented
-  | 'cache'
-  // deprecated
-  | 'moduleIds'
+  'cache'
 >
 
-export let parser = acorn.Parser.extend(importAssertions)
+export type RollupParseFunc = (
+  input: string,
+  options?: { allowReturnOutsideFunction?: boolean },
+) => AstNode
+
+export const getRollupParseFunc = async (): Promise<RollupParseFunc> => {
+  let rollupParse!: RollupParseFunc
+  await rollup({
+    input: 'dummy',
+    plugins: [
+      {
+        name: 'get-parse',
+        resolveId(id) {
+          return id
+        },
+        load(id) {
+          return ''
+        },
+        transform(code, id) {
+          rollupParse = this.parse
+        },
+      },
+    ],
+  })
+  return rollupParse
+}
```

Comment:
Neat trick! Maybe we can check with Lukas if this can be exported directly from Rollup so we don't have to hack this. The parse function seems to be rather self-contained. https://github.com/rollup/rollup/blob/4e562e52db3f55832de17ce02fabcc05c292861f/src/Graph.ts#L130-L132

Reply:
@lukastaegert We would like to use `this.parse` from rollup in the dev plugin pipeline. Would you consider exporting the function from rollup?

Reply:
That should be no problem as it is essentially a stateless function. Will see if I can manage to squeeze that in.

For the long term, I am thinking about a better API that only returns Proxies to generate AST nodes as needed and has advanced features for AST traversal, but not in the near future. Once this comes to pass, we can think about exposing this as well.

Reply:
Ok, here you go: rollup/rollup#5169
If I get a thumbs up in the next two hours, we might put it into the initial release of Rollup 4.

Reply:
Ah, I will just merge it anyway. Off to a last-minute beta release.

Reply:
Thanks! I was just looking at it. The interface looks good to me 👍